File: app/src/main/java/com/ryanrvldo/moviecatalogue/data/source/model/TvShow.java
package com.ryanrvldo.moviecatalogue.data.source.model;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.ryanrvldo.moviecatalogue.data.source.remote.response.ContentRatingResponse;
import com.ryanrvldo.moviecatalogue.data.source.remote.response.CreditsResponse;
import com.ryanrvldo.moviecatalogue.data.source.remote.response.ImageResponse;
import com.ryanrvldo.moviecatalogue.data.source.remote.response.SimilarResponse;
import com.ryanrvldo.moviecatalogue.data.source.remote.response.VideosResponse;

import java.util.List;

import lombok.Data;

@Data
@Entity(tableName = "favorite_tv")
public class TvShow {
    @PrimaryKey
    @Expose
    private int id;

    @SerializedName("name")
    @Expose
    private String title;

    @Expose
    private String overview;

    @ColumnInfo(name = "poster_path")
    @SerializedName("poster_path")
    @Expose
    private String posterPath;

    @ColumnInfo(name = "release_date")
    @SerializedName("first_air_date")
    @Expose
    private String releaseDate;

    @SerializedName("vote_average")
    @Expose
    private float rating;

    @Ignore
    @SerializedName("vote_count")
    @Expose
    private int ratingVotes;

    @SerializedName("backdrop_path")
    @Expose
    private String backdrop;

    @Ignore
    @Expose
    private List<Genre> genres;

    @Ignore
    @Expose
    private List<Season> seasons;

    @Ignore
    @Expose
    private ImageResponse images;

    @Ignore
    @SerializedName("content_ratings")
    @Expose
    private ContentRatingResponse contentRatingResponse;

    @Ignore
    @Expose
    private VideosResponse videos;

    @Ignore
    @Expose
    private CreditsResponse credits;

    @Ignore
    @Expose
    private SimilarResponse similar;

    public TvShow(int id, String title, String overview, String posterPath, String releaseDate, float rating, String backdrop) {
        this.id = id;
        this.title = title;
        this.overview = overview;
        this.posterPath = posterPath;
        this.releaseDate = releaseDate;
        this.rating = rating;
        this.backdrop = backdrop;
    }
}

File: app/src/main/java/com/ryanrvldo/moviecatalogue/data/model/TvShow.kt
package com.ryanrvldo.moviecatalogue.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.ryanrvldo.moviecatalogue.data.remote.response.*

@Entity(tableName = "favorite_tv")
data class TvShow(
    @Expose @SerializedName("id") @PrimaryKey val id: Int,
    @Expose @SerializedName("name") val title: String,
    @Expose val overview: String,
    @Expose @SerializedName("poster_path") @ColumnInfo(name = "poster_path") val posterPath: String,
    @Expose @SerializedName("first_air_date") @ColumnInfo(name = "release_date") val releaseDate: String,
    @Expose @SerializedName("vote_average") val rating: Float,
    @Expose @SerializedName("backdrop_path") val backdrop: String
) {
    @Ignore
    @Expose
    @SerializedName("vote_count")
    var ratingVotes: Int = 0

    @Ignore
    @Expose
    var genres: List<Genre>? = null

    @Ignore
    @Expose
    var seasons: List<Season>? = null

    @Ignore
    @Expose
    var images: ImageResponse? = null

    @Ignore
    @Expose
    @SerializedName("content_ratings")
    var contentRatingResponse: ContentRatingResponse? = null

    @Ignore
    @Expose
    var videos: VideosResponse? = null

    @Ignore
    @Expose
    var credits: CreditsResponse? = null

    @Ignore
    @Expose
    var similar: SimilarResponse? = null
}